src/admin/index.js

```diff
@@ -772,6 +775,280 @@ module.exports = ({
     })
   }
 
+  /**
+   * @param {array} ACL
+   * @return {Promise<void>}
+   */
+  const createAcls = async ({ acl }) => {
+    if (!acl || !Array.isArray(acl)) {
+      throw new KafkaJSNonRetriableError(`Invalid ACL array ${acl}`)
+    }
+    if (acl.length === 0) {
+      throw new KafkaJSNonRetriableError(`Empty ACL array`)
+    }
+
+    // Validate principal
+    if (acl.filter(({ principal }) => typeof principal !== 'string').length > 0) {
```

Comment:
A simpler way of expressing this is `acl.some(({ principal }) => typeof principal !== 'string')`. Same goes for the other validations.